src/cmp.rs

```diff
@@ -1,38 +1,98 @@
 use crate::iter::{iterate_lexical, iterate_lexical_only_alnum};
-use core::cmp::Ordering;
-
-macro_rules! cmp_ascii_digits {
-    (first_digits($lhs:ident, $rhs:ident), iterators($iter1:ident, $iter2:ident)) => {
-        let mut n1 = ascii_to_u64($lhs);
-        let mut n2 = ascii_to_u64($rhs);
-        loop {
-            match (
-                $iter1.peek().copied().filter(|c| c.is_ascii_digit()),
-                $iter2.peek().copied().filter(|c| c.is_ascii_digit()),
-            ) {
-                (Some(lhs), Some(rhs)) => {
-                    n1 = n1 * 10 + ascii_to_u64(lhs);
-                    n2 = n2 * 10 + ascii_to_u64(rhs);
-                    let _ = $iter1.next();
-                    let _ = $iter2.next();
-                }
-                (Some(_), None) => return Ordering::Greater,
-                (None, Some(_)) => return Ordering::Less,
-                (None, None) => {
-                    if n1 != n2 {
-                        return n1.cmp(&n2);
+use core::{
+    cmp::Ordering,
+    iter::Peekable,
+};
+
+fn cmp_ascii_digits(lhs: &mut Peekable<impl Iterator<Item=char>>, rhs: &mut Peekable<impl Iterator<Item=char>>) -> Option<Ordering> {
```

Comment:
I'm not big on macros that change control-flow outside their scope, like `cmd_ascii_digits!()` did via `return`. I've opted to replace it with a regular function, but that's personal preference. If you liked the macro with `return` for the caller better, I can change it back.